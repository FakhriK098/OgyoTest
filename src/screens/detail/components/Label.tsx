import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { ILabelProps } from 'src/types/user';

const Label = ({ title, value }: ILabelProps) => {
  if (!value) return <></>;
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{`${title}:`}</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  );
};

export default Label;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginBottom: 8,
    width: '90%',
  },
  title: {
    fontSize: 14,
    fontWeight: '600',
    marginRight: 4,
  },
  value: {
    fontSize: 14,
    fontWeight: '400',
    marginRight: 4,
  },
});
