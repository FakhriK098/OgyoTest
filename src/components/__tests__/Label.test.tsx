import React from 'react';
import { render } from '@testing-library/react-native';
import Label from '../Label';

describe('Label', () => {
  it('should render title and value correctly', () => {
    const { getByText } = render(<Label title="Followers" value="100" />);

    expect(getByText('Followers:')).toBeTruthy();
    expect(getByText('100')).toBeTruthy();
  });

  it('should render nothing with empty value', () => {
    const { toJSON } = render(<Label title="Bio" />);

    expect(toJSON()).toBeNull();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<Label title="Test Label" value="Test Value" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
