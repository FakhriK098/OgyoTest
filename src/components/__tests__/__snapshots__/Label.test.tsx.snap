// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should match snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginBottom": 8,
      "width": "90%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "600",
        "marginRight": 4,
      }
    }
  >
    Test Label:
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "400",
        "marginRight": 4,
      }
    }
  >
    Test Value
  </Text>
</View>
`;
